### PR TITLE
Contact popup

### PR DESCRIPTION
**Pull request by nclayt0n**
Created a contact widget component

### CODE REVIEW

**Comment by nclayt0n** [comment]
https://github.com/FindawayWorld/gateway/wiki/Contact-Widget

File: src/components/ContactWidget.jsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                        } else if (
+                            !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(
+                                values.email
+                            )
+                        ) {
+                            errors.email = 'Invalid email address';
+                        }
+                        return errors;
+                    }}
+                    onSubmit={(values, { setSubmitting }) => {
```
Comment (cwardzala) [inline src/components/ContactWidget.jsx:63]: Move this to a function outside `return`. `const handleSend = () => {...}`

File: src/components/ContactWidget.jsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                        id: props.id,
+                        email: '',
+                        orgType: '',
+                        name: '',
+                        orgName: '',
+                        notes: '',
+                        phone: '',
+                        city: '',
+                        state: '',
+                        key: 'contactWidgetFormik',
```
Comment (cwardzala) [inline src/components/ContactWidget.jsx:48]: what is `key` for?

File: src/components/ContactWidget.jsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+                    <span className="ml-6 span">
+                        <FaChevronDown
+                            className={classnames('', { hide: !isOpen })}
+                            size={14}
+                        />
+                    </span>
+                </button>
+                <Formik
+                    initialValues={{
+                        id: props.id,
```
Comment (cwardzala) [inline src/components/ContactWidget.jsx:39]: what is `id` for? avoid setting values for items the user cannot change. You can always add more items to the axios data when sending.

File: src/components/ContactWidget.jsx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+                                        id="state"
+                                        type="text"
+                                        label="State"
+                                        labelClass="mb-1"
+                                        value={values.state}
+                                        error={errors.state}
+                                        onChange={handleChange}
+                                    >
+                                        <option value="">--</option>
+                                        {fiftyStates.map((state, idx) => (
```
Comment (cwardzala) [inline src/components/ContactWidget.jsx:208]: what happens if the user is outside the US? Maybe think about this being a text box to collect their state/province.

**Pull request closed without merging**